gslb/cache/controller_obj_cache.go

```diff
@@ -421,26 +421,57 @@ func parseDescription(description string) ([]string, error) {
 	return objList, nil
 }
 
-func ParsePoolAlgorithmSettings(gsPool models.GslbPool) *gslbalphav1.PoolAlgorithmSettings {
-	if gsPool.Algorithm != nil {
+func ParsePoolAlgorithmSettingsFromPool(gsPool models.GslbPool) *gslbalphav1.PoolAlgorithmSettings {
+	return ParsePoolAlgorithmSettings(gsPool.Algorithm, gsPool.FallbackAlgorithm, gsPool.ConsistentHashMask)
+}
+
+func ParsePoolAlgorithmSettings(algorithm *string, fallbackAlgorithm *string, consistentHashMask *int32) *gslbalphav1.PoolAlgorithmSettings {
+	if algorithm != nil {
 		return nil
 	}
-	pa := gslbalphav1.PoolAlgorithmSettings{LBAlgorithm: *gsPool.Algorithm}
-	if gsPool.FallbackAlgorithm != nil {
+	pa := gslbalphav1.PoolAlgorithmSettings{LBAlgorithm: *algorithm}
+	if fallbackAlgorithm != nil {
 		gfa := gslbalphav1.GeoFallback{
-			LBAlgorithm: *gsPool.FallbackAlgorithm,
+			LBAlgorithm: *fallbackAlgorithm,
 		}
-		if gsPool.ConsistentHashMask != nil {
-			hashMask := int(*gsPool.ConsistentHashMask)
+		if consistentHashMask != nil {
```

Comment:
both the checks ( next else ) can be merged irrespective of fallbackAlgorithm?

Reply:
I don't think so. `consistentHashMask` has to be checked in different contexts. If its in the context of `fallbackAlgorithm`, then we assign it to the geoFallback struct, or else, we assign it to the normal algorithm settings.

However, I think there's a bug here, where we are not assigning the `geoFallback` to `pa`. Will fix it.